include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -429,6 +430,8 @@ struct SLightSample
     }
     scalar_type getNdotL2() NBL_CONST_MEMBER_FUNC { return NdotL2; }
 
+    bool isValid() NBL_CONST_MEMBER_FUNC { return !hlsl::all<vector<bool, 3> >(hlsl::glsl::equal(L.getDirection(), hlsl::promote<vector3_type>(0.0))); }
```

Comment:
why not use `glsl::notEqual` and not-unary-not it? I mean SPIR-V `OpFOrdNotEqual` exists so that intrinsic should probably get made